src/safeds/data/tabular/containers/_time_series.py

```diff
@@ -906,3 +906,64 @@ def plot_lagplot(self, lag: int) -> Image:
         plt.close()  # Prevents the figure from being displayed directly
         buffer.seek(0)
         return Image.from_bytes(buffer.read())
+
+    def plot_moving_average(
+        self,
+        window_size: int,
+        feature_name: str | None = None,
+    ) -> Image:
+        """
+        Plot the moving average for the target column.
+
+        Parameters
+        ----------
+        window_size:
+            The size of the windows, which the average gets calculated for
+
+        feature_name:
+            The name of the column which will be used to calculate the moving average, if None the target column will be taken
+
+        Returns
+        -------
+        plot:
+            The moving avereage plot and the normal plot as an image.
+
+        Raises
+        ------
+        NonNumericColumnError
+            If the time series targets contains non-numerical values.
+
+        UnknownColumnNameError
+            If the time series doesn't contain the given column name
+
+        Examples
+        --------
+                >>> from safeds.data.tabular.containers import TimeSeries
+                >>> table = TimeSeries({"time":[1, 2], "target": [3, 4], "feature":[2,2]}, target_name= "target", time_name="time", feature_names=["feature"], )
+                >>> image = table.plot_moving_average(window_size = 2)
+
+        """
+        if feature_name is None or feature_name == self.target.name:
+            series = self.target._data
+            feature_name = self.target.name
+        else:
+            if feature_name not in self.column_names:
+                raise UnknownColumnNameError([feature_name])
+            series = self._data[feature_name]
+        if not self.get_column(feature_name).type.is_numeric():
+            raise NonNumericColumnError("This time series plotted column contains non-numerical columns.")
+
+        # create moving average series
+        series_mvg = series.rolling(window_size).mean()
+
+        # plot both series and put them together
+        ax_temp = series_mvg.plot()
+        ax = series.plot(ax=ax_temp)
+        ax.legend(labels=["moving_average", feature_name])
+
+        fig = ax.figure
+        buffer = io.BytesIO()
+        fig.savefig(buffer, format="png")
+        plt.close()  # Prevents the figure from being displayed directly
+        buffer.seek(0)
+        return Image.from_bytes(buffer.read())
```

Comment:
We are using this for all plots. Can we extract that into an internal function`_create_image_for_plot(fig: ?) -> Image` in a file `src/safeds/_utils/_plotting.py`? Also add a reexport of the function to `src/safeds/_utils/__init__.py`.

Reply:
added